src/MSBuildLocator/DotNetSdkLocationHelper.cs

```diff
@@ -102,14 +102,18 @@ private static IEnumerable<string> GetDotNetBasePaths(string workingDirectory)
                 string filePath = Path.Combine(dir, exeName);
                 if (File.Exists(Path.Combine(dir, exeName)))
                 {
-                    dotnetPath = filePath;
-                    break;
+                    filePath = Path.GetDirectoryName(isWindows ? filePath : realpath(filePath) ?? filePath);
+                    if (File.Exists(Path.Combine(filePath, exeName)))
```

Comment:
Are we now checking the same file twice on Windows?

Reply:
Yes, so it could be optimized somewhat, but I care very little about performance on this code path, since it should only execute once per SDK you have installed...so like twice or thrice for the average user.